wagtail/admin/templatetags/wagtailadmin_tags.py

```diff
@@ -446,6 +446,9 @@ def paginate(context, page, base_url="", page_key="p", classname=""):
 @register.inclusion_tag("wagtailadmin/shared/buttons.html", takes_context=True)
 def page_listing_buttons(context, page, user):
     next_url = context["request"].path
+    # When searching a page title, the next_url gets index results path
+    if next_url.endswith('/results/'):
+        next_url = next_url.replace('/results/', '/')
```

Comment:
I'm not sure relying on matching `/results/` is the best approach here. It's somewhat naive and hard to keep track of. We might be better off adding an optional parameter to the tag for the `next_url` which defaults to `context["request"].path` (or having a specific context variable for `next_url`), then passing the correct `next_url` explicitly in the results view.

Reply:
Agreed - have opened a new PR following this approach at #11191. It's slightly fiddly due to the need to pass the parameter via the table component, but I think this is the right way to go.